### PR TITLE
Extendable LoggerNode att data structure

### PR DESCRIPTION
**Pull request by vsevel**
Replace attX attributes by one array extendable datastructure.
Workaround for https://github.com/quarkusio/quarkus/issues/26401

### CODE REVIEW

**Comment by vsevel** [comment]
needs some additional tests.

**Comment by vsevel** [comment]
added some tests. to me it is ready to review @dmlloyd

File: src/main/java/org/jboss/logmanager/LoggerNode.java [hunk 1]
```diff
@@ -455,34 +417,25 @@ <V> V attachIfAbsent(final Logger.AttachmentKey<V> key, final V value) {
...
             lock.unlock();
         }
-        throw new IllegalStateException("Maximum number of attachments exceeded");
+    }
+
+    private <V> void extend(Logger.AttachmentKey<V> key) {
+        if(key.getId() >= attArray.length) {
+            Object[] newAttArray = new Object[key.getId() + 1];
+            System.arraycopy(attArray, 0, newAttArray, 0, attArray.length);
+            attArray = newAttArray;
```
Comment (dmlloyd) [inline src/main/java/org/jboss/logmanager/LoggerNode.java]: Use `Arrays.copyOf(attArray, key.getId() + 1)` instead here.

**Review by dmlloyd — APPROVED** [review]
Only one minor complaint, but otherwise looks OK.

**Comment by vsevel** [comment]
hello @dmlloyd any news on this?

**Comment by dmlloyd** [comment]
Looks good. Sorry I didn't get to it sooner, yesterday was a holiday here.

**Comment by vsevel** [comment]
hi @dmlloyd do you intend to do a release at some point?

**Comment by vsevel** [comment]
hi @dmlloyd has it been ever released? or fixed some other way? I did not see any changes on https://github.com/quarkusio/quarkus/issues/26401 ...

**Comment by dmlloyd** [comment]
This change was included in a release; an upstream pull request would be required to update the component in Quarkus and I haven't gotten around to it (anyone can do it though).

**Comment by vsevel** [comment]
ah cool! so I suppose it is in `1.1.1`? I will do the bump in Quarkus, np.

**Pull request merged**